mod/gc_elgg_sitemap/start.php

```diff
@@ -0,0 +1,380 @@
+
+	    /// renmove these pages so that it doesn't get crawled
+		elgg_unregister_page_handler('activity');
+		elgg_unregister_page_handler('dashboard');
+		elgg_unregister_menu_item('topbar', 'dashboard');
+		elgg_unregister_menu_item('site', 'activity');
+		elgg_unregister_menu_item('site', 'career');
+		elgg_unregister_menu_item('site', 'Help');
+		elgg_unregister_menu_item('site', 'newsfeed');
+		elgg_unregister_menu_item('site', 'communities');
+		elgg_unregister_menu_item('title2', 'new_folder');
+
+		/// list all entities
+		elgg_register_plugin_hook_handler('view_vars', 'object/elements/summary', 'elgg_entities_list_handler');
+		elgg_register_plugin_hook_handler('view_vars', 'object/elements/full', 'elgg_full_entities_view_handler');
+		elgg_register_plugin_hook_handler('view_vars', 'object/elements/thewire_summary', 'elgg_thewire_list_handler');
+		elgg_register_plugin_hook_handler('view_vars', 'group/elements/summary', 'elgg_entities_list_handler');
+		elgg_register_plugin_hook_handler('view_vars', 'page/components/image_block', 'elgg_sidebar_handler');
+
+		elgg_register_plugin_hook_handler('view', 'members/nav', 'elgg_members_menu_handler');
+		elgg_register_plugin_hook_handler('view', 'event_calendar/filter_menu', 'elgg_members_menu_handler');
+
+
+	    elgg_register_plugin_hook_handler('register','menu:owner_block','group_navigation_handler');
+		elgg_extend_view('groups/profile/summary', 'group_tabs', 600);
+
+		elgg_register_plugin_hook_handler('entity:url', 'group', 'redirect_group_url');
+		elgg_register_plugin_hook_handler('entity:url', 'object', 'redirect_content_url', 1);
+		elgg_register_plugin_hook_handler('view', 'output/longtext', 'strip_content_hyperlinks_handler');
+
+	}
+}
+
+
+function strip_content_hyperlinks_handler($hook, $type, $return, $params) {
+	
+	// pull the user agent string and/or user testing
+	$gsa_agentstring = strtolower(elgg_get_plugin_setting('gsa_agentstring','gc_fedsearch_gsa'));
+	$gsa_usertest = elgg_get_plugin_setting('gsa_test','gc_fedsearch_gsa');
+	if ($gsa_usertest) $current_user = elgg_get_logged_in_user_entity();
+
+	/*blog pages bookmarks file discussion*/
+	$filter_entity = array('blog', 'pages', 'discussion', 'file', 'bookmarks');
+	$context = get_context();
+
+	// only do it for the main content, comments will be left the way it is
+	$comment = new DOMDocument();
+	$comment->loadHTML($return);
+	$comment_block = $comment->getElementsByTagName('div');
+
+	if (strstr($return, 'data-role="comment-text"') !== false) {
+		return;
+	}
+
+	if (!empty(trim($comment_block)) || $comment_block === null)
+		$comment_text = $comment_block->item(0)->getAttribute('data-role');
+	else
+		$comment_text = "";
+
+	if (( strcmp($comment_text,'comment-text') == 0 || strcmp($comment_text, 'discussion-reply-text') == 0 ))
+		return;
+
+	if (!in_array($context, $filter_entity))
+		return;
+
+	$url = explode('/',$_SERVER['REQUEST_URI']);
+	$entity = get_entity($url[4]);
```

Comment:
I don't think there's any other way of pulling the entity out (it just contains a block of html) but i will have to modify so it's not hardcoded to extracting that specific block in the URL